blog/2024-10-29-vibration-analysis-python/index.mdx

```diff
@@ -0,0 +1,376 @@
+
+An interesting consequence of an ability to habituate is sensitivity to novelty: whenever a prediction cannot be made confidently or appears to be wrong, we immediately recognize the ongoing experience as new. This criterion of novelty can be used for both further learning and identifying unusual states of the system - detecting anomalies. The more complex connections and patterns of system behavior we - or an artificial model - have learned, the more subtle and non-obvious changes in the system can be noticed. Thus, AI-driven anomaly detection may be sensible to changes even before they are formalized and described, including changes we might neither expect nor be able to interpret well.
+
+However, the field of ML/DL applied to signal processing is so wide, that it barely can be covered in a single post.
+
+## Using ReductStore for selective data replication based on vibration analysis
+
+Even a preliminary vibration analysis, feasible as an edge-computing task, can be used as a basis for selective data replication and storage. With ReductStore, the basic scenario of selective data replication is described in [**this post**](https://www.reduct.store/blog/how-to-store-vibration-sensor-data). The basic scenario of selective replication, described there, is based on (editable) labels of records, offered by ReductStore: each record can be labeled with custom key-value pairs, both during creation or later (as labels of existing records remain editable). In a most simple case, one can compute vibration features for each record being created, label the records dependent on those features, and later use specific labels to select records, i.e. copy them to a longer-life storage for retention.
+
+Let's review such a use case in more detail. For simplicity, let's consider a machine under vibration monitoring that has a single well-known stationary regime of operation, represented by a reference record. If we are interested in machine health diagnostics, we may be interested in saving only records that significantly differ from this state. We can formulate criteria of difference as a metric, measure that difference by computing the related metric of each recent record against the reference one, and label records by difference when saving to ReductStore short-term storage bucket, configured to serve as a ring buffer, limited by size quota, and finally create a conditional replication task by means of ReductStore (as described [**here**](https://www.reduct.store/docs/guides/data-replication)), that selects records by labels and regularly copies them to long-term storage bucket (see also [**this repository**](https://github.com/reductstore/reduct-vibration-example) for more details).
```

Comment:
For search engines, it's good to have descriptive anchor texts, and with docusaurus we can use relative path to the site. So it would something like: 

"as described [**in the data replication guide**](/docs/guides/data-replication)"